src/main/java/ru/l1ttleO/boyfriend/commands/util/Sender.java

```diff
@@ -0,0 +1,66 @@
+package ru.l1ttleO.boyfriend.commands.util;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import org.jetbrains.annotations.NotNull;
+import ru.l1ttleO.boyfriend.Boyfriend;
+import ru.l1ttleO.boyfriend.I18n.BotLocale;
+import ru.l1ttleO.boyfriend.Utils;
+import ru.l1ttleO.boyfriend.settings.GuildSettings;
+
+import static ru.l1ttleO.boyfriend.I18n.tl;
+
+public abstract class Sender {
+    public abstract void reply(final @NotNull CharSequence text);
+
+    public void replyTl(final @NotNull String key, final Object... args) {
+        this.reply(tl(key, this.getLocale(), args));
+    }
+
+    public abstract BotLocale getLocale();
+
+    public void update() {
+    }
+
+    public static class MessageSender extends Sender {
+        public final @NotNull Message message;
+        private @NotNull BotLocale locale;
+
+        public MessageSender(final @NotNull Message message, final BotLocale locale) {
```

Comment:
```suggestion
        public MessageSender(final @NotNull Message message, final @NotNull BotLocale locale) {
```
`Constructor parameter for @NotNull field might be annotated @NotNull itself` [intellij]